### PR TITLE
yomm2: add version 1.5.2

### PR DESCRIPTION
**Pull request by jll63**
Specify library name and version:  **yomm2/1.5.2**

Updating to the latest version of YOMM2, which I just published. Also:
* I think that the patch is not necessary anymore.
* I changed `test_package.cpp` to make it _call_ a method, not just build method tables (actually, this did nothing in the absence of any declared method).
---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/yomm2/all/conanfile.py [hunk 1]
```diff
@@ -79,12 +79,7 @@ def generate(self):
         tc.variables["YOMM2_SHARED"] = not bool(self.options.header_only)
         tc.generate()
 
-    def _patch_sources(self):
-        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
-        replace_in_file(self, cmakelists, "add_subdirectory(docs.in)", "")
-
```
Comment (franramirez688) [inline recipes/yomm2/all/conanfile.py:85]: This patch still makes sense for the 1.5.1 version, so it'd be better to distinguish between versions. Something like:

```python
        if Version(self.version) <= "1.5.1":
            cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
            replace_in_file(self, cmakelists, "add_subdirectory(docs.in)", "")
```

Where `Version` is imported as `from conan.tools.scm import Version`

**Review by franramirez688 — COMMENTED** [review]
Hi @jll63 - Thanks a lot for your PR! 👏 

We are adding you to the list of authorized users asap 😁 

Your PR looks great! I only have a small suggestion related to the patch.

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by jll63** [comment]
Should I click "Update branch"? Or you just merge?

**Comment by franramirez688** [comment]
> Should I click "Update branch"? Or you just merge?

No, don't worry. Our Bot will automatically merge it ASAP 😉
"Update branch" is a shortcut to update your branch with the master one and push the changes to relaunch the CI.

Thanks for asking @jll63

**Pull request merged**